File: ove/flutter/lib/domain/session_state.dart
enum SessionStep {
  idle,
  recording,
  transcribing,
  restating,
  followUp,
  beliefSelection,
  generating,
  report,
  complete;

  bool get isTerminal => this == complete;

  bool get isLoading =>
      this == transcribing || this == restating || this == generating;
}

File: ove/web/lib/session-state.ts
export type SessionStep =
  | 'idle'
  | 'inputting'
  | 'restating'
  | 'followUp'
  | 'empathy'
  | 'beliefHypothesis'
  | 'beliefSelection'
  | 'generating'
  | 'report'
  | 'complete';
